### PR TITLE
Point set shape detection: Plane regularization

### PR DESCRIPTION
**Pull request by sgiraudot**
Small feature: https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Plane_regularization
* Not approved yet
* Partially tested

### CODE REVIEW

**Comment by lrineau** [comment]
There are warnings on Windows:
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-1.shtml#Point_set_shape_detection_3

**Comment by lrineau** [comment]
There are compilation errors in plugins of Polyhedron demo:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-Ic-1/Polyhedron_Demo/TestReport_afabri_x64_Cygwin-Windows8_MSVC2013-Release-64bits.gz
> test\Polyhedron_Demo\Plugins\Point_set\Point_set_shape_detection_plugin.cpp(11) : fatal error C1083: Cannot open include file: 'CGAL/Plane_regularization.h': No such file or directory

**Comment by sgiraudot** [comment]
All should be fixed now.

(Sorry about the Polyhedron demo part, I always forget to propagate changes of API into the related plugin...)

**Comment by sgiraudot** [comment]
This PR was pre-approved by @palliez (https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Plane_regularization). I took his latest remarks into account.

I also fixed some bugs and added an executable for the testsuite.

**Comment by sgiraudot** [comment]
There were 2 warnings of implicit conversion of double to FT which I corrected. Everything else seems to work fine (the errors on the `IntelCompiler-17` platform do not seem related to this PR).

**Comment by sloriot** [comment]
Wiki updated

File: Installation/changes.html [hunk 1]
```diff
@@ -269,6 +269,13 @@ <h2 id="release4.8">Release 4.8 </h2>
       <code>CGAL::write_ply_points()</code>
       and <code>CGAL::write_ply_points_and_normals()</code>.</li>
   </ul>
+  <h3>Point Set Shape Detection</h3>
+  <ul>
+    <li>New post-processing
+      algorithm: <code>CGAL::regularize_planes()</code>. This allows the user
+      to favor parallelism, orthogonality, coplanarity and/or axial
+      symmetry between detected planes.</li>
+  </ul>
```
Comment (lrineau) [inline Installation/changes.html]: The PR is merged in `master` (4.9) but the modification to the `changes.html` is in the section about CGAL-4.8!
@sloriot @sgiraudot

Comment (lrineau) [inline Installation/changes.html]: Done by PR #1062.

**Pull request merged**